dietpi/dietpi-software

```diff
@@ -5647,15 +5648,14 @@ _EOF_
 
 		if To_Install 195 # youtube-dl
 		then
-			aDEPS=('python3')
-			Download_Install 'https://yt-dl.org/downloads/latest/youtube-dl' /usr/local/bin/youtube-dl
-			G_EXEC chmod +x /usr/local/bin/youtube-dl
+			# Remove obsolete youtube-dl and youtube-dl-py2 we used previously
+			G_EXEC rm -f /usr/local/bin/youtube-dl{,-py2}
 
-			# youtube-dl supports Python 2 and Python 3, but its shebang calls Python 2, else fails: https://github.com/ytdl-org/youtube-dl/issues/27649
-			G_EXEC sed -i '1s/python$/python3/' /usr/local/bin/youtube-dl
+			aDEPS=('python3')
+			Download_Install 'https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp' /usr/local/bin/yt-dlp
+			G_EXEC chmod +x /usr/local/bin/yt-dlp
 
-			# Remove obsolete /usr/local/bin/youtube-dl-py2 we used previously
-			[[ -f '/usr/local/bin/youtube-dl-py2' ]] && G_EXEC rm /usr/local/bin/youtube-dl-py2
+			G_EXEC ln -s /usr/local/bin/yt-dlp /usr/local/bin/youtube-dl
```

Comment:
Skip the removal before the download, so users as not left without any binary if the download fails for some reason. Instead use `-f` option to overwrite any existing file or symlink here:
```suggestion
			G_EXEC ln -sf /usr/local/bin/yt-dlp /usr/local/bin/youtube-dl
```
The py2 removal could/should then stay where it was.

Reply:
Done.